06.bowling_object/game.rb

```diff
@@ -0,0 +1,41 @@
+#!/usr/bin/env ruby
+# frozen_string_literal: true
+
+require './shot'
+require './frame'
+
+class Game
+  def initialize(score)
+    @frames = parse_frames(score)
+  end
+
+  def calc_total
+    @frames.each.sum do |frame|
+      frame.score(@frames)
+    end
+  end
+
+  private
+
+  def parse_frames(score)
+    marks = score.split(',')
+    frames = (1..9).map do |i|
+      if strike?(marks[0])
```

Comment:
21行目の時点でShotインスタンスの配列を作ってしまいましょう。
その上で、

```suggestion
      if shots[0].strike?
```

のようにして、Shotクラスにストライクかどうかの判定をさせるのがいいと思います。

また、Frame.newするときもShotの配列を渡すようにしてみてください。

Reply:
Shotのインスタンスの配列を作り、Shotクラスに判定させるようにしました